docs/src/main/sphinx/develop/client-protocol.rst

```diff
@@ -0,0 +1,257 @@
+  * - ``data``
+    - The ``data`` attribute contains a list of the rows returned by the
+      query request.  Each row is itself a list that holds values of the
+      columns in the row, in the order specified by the ``columns``
+      attribute.
+  * - ``updateType``
+    - A human-readable string representing the operation.  For a
+      ``CREATE TABLE`` request, the ``updateType`` will be
+      "CREATE TABLE"; for ``SET SESSION`` it will be "SET SESSION"; etc.
+  * - ``error``
+    - If query failed, the ``error`` attribute will contain JSON for a
+      ``QueryError`` object.  That object contains a ``message``, an
+      ``errorCode`` and other information about the error.  See the
+      ``QueryError`` class for more details.
+
+
+``QueryResults`` diagnostic attributes
+======================================
+
+These ``QueryResults`` data members may be useful in tracking down problems:
+
+.. list-table:: ``QueryResults diagnostic attributes``
+  :widths: 20, 20, 40
+  :header-rows: 1
+
+  * - Attribute
+    - Type
+    - Description
+  * - ``queryError``
+    - ``QueryError``
+    - Non-null only if the query resulted in an error.
+  * - ``failureInfo``
+    - ``FailureInfo``
+    - ``failureInfo`` has detail on the reason for the failure, including
+      a stack trace, and ``FailureInfo.errorLocation``, providing the
+      query line number and column number where the failure was detected.
+  * - ``warnings``
+    - ``List<TrinoWarning>``
+    - A usually-empty list of warnings.
+  * - ``statementStats``
+    - ``StatementStats``
+    - A class containing statistics about the query execution.  Of
+      particular interest is ``StatementStats.rootStage``, of type
+      ``StageStats``, providing statistics on the execution of each of
+      the stages of query processing.
+
+Client request headers
+======================
+
+This table lists all supported client request headers.  Many of the
+headers can are updated in the client by response headers, and supplied
+in subsequent requests, just like browser cookies.
+
+.. list-table:: Client request headers
+  :widths: 30, 50
+  :header-rows: 1
+
+  * - Header Name
+    - Description
+  * - ``X-Trino-User``
+    - Specifies the session user; must be supplied with every
+      request to ``/v1/statement``.
+  * - ``X-Trino-Source``
+    - For reporting purposes, this supplies the name of the software
+      that submitted the query.
+  * - ``X-Trino-Catalog``
+    - The catalog to be used when running the query.  Set by response
+      header ``X-Trino-Set-Catalog``.
+  * - ``X-Trino-Schema``
+    - The schema to be used when running the query.  Set by response
+      header ``X-Trino-Set-Schema``.
+  * - ``X-Trino-Time-Zone``
+    - The timezone to be used when running the query, which by default
+      is the timezone of the Trino engine.
+  * - ``X-Trino-Language``
+    - The language to be used when running the query and formatting
```

Comment:
The language to use when processing the query and formatting results.

Reply:
Changed.